In [5]:
#importing neccessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


# Defining Problem Statement and Analysing basic metrics:

Netflix is one of the most popular media and video streaming platforms. They have over 8000 movies or tv shows available on their platform, as of mid-2021, they have over 200M Subscribers globally.

The particular business case focuses on the Netflix show data and provides insightful information on 8807 shows

Analyzing the data and generating insights helps Netflix decide which type of shows/movies to produce and how to grow the business.

In [6]:
#load the dataset
netflix_data=pd.read_csv('netflix.csv')

In [7]:
#showing top 10 rows
netflix_data.head(10)

show_id     type                             title  \
0      s1    Movie              Dick Johnson Is Dead   
1      s2  TV Show                     Blood & Water   
2      s3  TV Show                         Ganglands   
3      s4  TV Show             Jailbirds New Orleans   
4      s5  TV Show                      Kota Factory   
5      s6  TV Show                     Midnight Mass   
6      s7    Movie  My Little Pony: A New Generation   
7      s8    Movie                           Sankofa   
8      s9  TV Show     The Great British Baking Show   
9     s10    Movie                      The Starling   

                        director  \
0                Kirsten Johnson   
1                            NaN   
2                Julien Leclercq   
3                            NaN   
4                            NaN   
5                  Mike Flanagan   
6  Robert Cullen, José Luis Ucha   
7                   Haile Gerima   
8                Andy Devonshire   
9                 Theodore Melfi   

                                                cast  \
0                                                NaN   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...   
3                                                NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...   
5  Kate Siegel, Zach Gilford, Hamish Linklater, H...   
6  Vanessa Hudgens, Kimiko Glenn, James Marsden, ...   
7  Kofi Ghanaba, Oyafunmike Ogunlano, Alexandra D...   
8  Mel Giedroyc, Sue Perkins, Mary Berry, Paul Ho...   
9  Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...   

                                             country          date_added  \
0                                      United States  September 25, 2021   
1                                       South Africa  September 24, 2021   
2                                                NaN  September 24, 2021   
3                                                NaN  September 24, 2021   
4                                              India  September 24, 2021   
5                                                NaN  September 24, 2021   
6                                                NaN  September 24, 2021   
7  United States, Ghana, Burkina Faso, United Kin...  September 24, 2021   
8                                     United Kingdom  September 24, 2021   
9                                      United States  September 24, 2021   

   release_year rating   duration  \
0          2020  PG-13     90 min   
1          2021  TV-MA  2 Seasons   
2          2021  TV-MA   1 Season   
3          2021  TV-MA   1 Season   
4          2021  TV-MA  2 Seasons   
5          2021  TV-MA   1 Season   
6          2021     PG     91 min   
7          1993  TV-MA    125 min   
8          2021  TV-14  9 Seasons   
9          2021  PG-13    104 min   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   
5                 TV Dramas, TV Horror, TV Mysteries   
6                           Children & Family Movies   
7   Dramas, Independent Movies, International Movies   
8                       British TV Shows, Reality TV   
9                                   Comedies, Dramas   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...  
5  The arrival of a charismatic young priest brin...  
6  Equestria's divided. But a bright-eyed hero be...  
7  On a photo shoot in Ghana, an American model s...  
8  A talented 

# Basic Metrics

First examine the basic metrics like shape and some basic statistics.

In [8]:
dataset_shape=netflix_data.shape
dataset_shape




(8807, 12)

In [9]:
dataset_dtypes=netflix_data.dtypes
dataset_dtypes

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object

# Observations-
The dataset contains 8807 rows and 12 attributes.

The dataset consists of 8,807 entries with 12 attributes:

show_id: Unique ID for every Movie / TV show
type: Identifier — A Movie or TV Show
title: Title of the Movie / TV Show
director: Director of the Movie
cast: Actors involved in the movie/show
country: The country where the movie/show was produced
date_added: Date it was added on Netflix
release_year: Actual Release year of the movie/show
rating: TV Rating of the movie/show
duration: Total Duration — in minutes or number of seasons
listed_in: Genre
description: The summary description

In [10]:
#Analysing basic metric for categorical  and numerical columns
basic_metric_cat_num=netflix_data.describe(include='all')
basic_metric_cat_num

show_id   type                 title       director  \
count     8807   8807                  8807           6173   
unique    8807      2                  8807           4528   
top         s1  Movie  Dick Johnson Is Dead  Rajiv Chilaka   
freq         1   6131                     1             19   
mean       NaN    NaN                   NaN            NaN   
std        NaN    NaN                   NaN            NaN   
min        NaN    NaN                   NaN            NaN   
25%        NaN    NaN                   NaN            NaN   
50%        NaN    NaN                   NaN            NaN   
75%        NaN    NaN                   NaN            NaN   
max        NaN    NaN                   NaN            NaN   

                      cast        country       date_added  release_year  \
count                 7982           7976             8797   8807.000000   
unique                7692            748             1767           NaN   
top     David Attenborough  United States  January 1, 2020           NaN   
freq                    19           2818              109           NaN   
mean                   NaN            NaN              NaN   2014.180198   
std                    NaN            NaN              NaN      8.819312   
min                    NaN            NaN              NaN   1925.000000   
25%                    NaN            NaN              NaN   2013.000000   
50%                    NaN            NaN              NaN   2017.000000   
75%                    NaN            NaN              NaN   2019.000000   
max                    NaN            NaN              NaN   2021.000000   

       rating  duration                     listed_in  \
count    8803      8804                          8807   
unique     17       220                           514   
top     TV-MA  1 Season  Dramas, International Movies   
freq     3207      1793                           362   
mean      NaN       NaN                           NaN   
std       NaN       NaN                           NaN   
min       NaN       NaN                           NaN   
25%       NaN       NaN                           NaN   
50%       NaN       NaN                           NaN   
75%       NaN       NaN                           NaN   
max       NaN       NaN                           NaN   

                                              description  
count                                                8807  
unique                                               8775  
top     Paranormal activity at a lush, abandoned prope...  
freq                                                    4  
mean                                                  NaN  
std                                                   NaN  
min                                                   NaN  
25%                                                   NaN  
50%                                                   NaN  
75%                                                   NaN  
max                                                   NaN

# Observations-
# For the numerical attribute release_year:

Count: 8,807 entries
Mean: Around the year 2014
Standard Deviation: Approximately 8.82 years
Minimum: Year 1925
25th Percentile (Q1): Year 2013
Median (50th Percentile): Year 2017
75th Percentile (Q3): Year 2019
Maximum: Year 2021

# For categorical attributes-
Type

Count: 8,807
Unique Values: 2 (Movie, TV Show)
Most Frequent: Movie
Frequency: 6,131
Country

Count: 7,976 (some missing values)
Unique Values: 748
Most Frequent: United States
Frequency: 2,818
Rating

Count: 8,803 (some missing values)
Unique Values: 17
Most Frequent: TV-MA
Frequency: 3,207





# Observations-
1.The average release year being around 2014,and the medioan year of 2017 suggests that Netflix has a lots of contents from recent decades.
2.Frequency of movies is greater than TV shows which indicates that more focus should be on movies content.
3.The United States has a larger proportion in creating content for Netflix followed by many other countries too.
4.Most frequent rating is 'TV-MA' suggesting a focus on mature audience.


In [14]:
#converting categorical attributes to category datatype
netflix_cat_attribute=['type','country','rating']
netflix_data[netflix_cat_attribute]=netflix_data[netflix_cat_attribute].astype("category")

#after conversion datatypes
after_conversion_dtypes=netflix_data.dtypes


# Data Types of All the Attributes (Before Conversion)
Most of the attributes are of object data type, except release_year, which is an int64.
# Conversion of Categorical Attributes to ‘Category’
The data types for type, country, and rating have been converted to category.

In [15]:
#missing value detection
netflix_missing_val=netflix_data.isnull().sum()
netflix_missing_val

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

# Observations-
1.columns like director=2634 missing values
cast=825 missing values
country=831 missing values
date_added=10 missing values
rating=4 missing values
duration=3 missing values



#Non graphical analysis-value counts and unique values detection-


In [18]:
#Non graphical analysis-value counts for key attributes-
value_counts_type=netflix_data['type'].value_counts()
value_counts_country=netflix_data['country'].value_counts().head(10)
value_counts_rating=netflix_data['rating'].value_counts()
value_counts_release_year=netflix_data['release_year'].value_counts().head(10)

#Non graphical analysis